ci-operator/step-registry/ipi/aws/pre/edge-zones/opt-in/ipi-aws-pre-edge-zones-opt-in-commands.sh

```diff
@@ -0,0 +1,104 @@
+        --filter Name=state,Values=available Name=zone-type,Values="${zone_type}" \
+        | jq -r '.AvailabilityZones[].ZoneName' | shuf | tail -n 1)
+    echo_date "Edge Zone selected: ${zone_name}"
+
+    zone_group_name=$(aws --region "$REGION" ec2 describe-availability-zones \
+        --all-availability-zones \
+        --filters Name=zone-name,Values="$zone_name" \
+        --query "AvailabilityZones[].GroupName" --output text)
+    echo_date "Zone Group discovered: ${zone_group_name}"
+
+    if [[ $(aws --region "$REGION" ec2 describe-availability-zones --all-availability-zones \
+            --filters Name=zone-type,Values="${zone_type}" Name=zone-name,Values="$zone_name" \
+            --query 'AvailabilityZones[].OptInStatus' --output text) == "opted-in" ]];
+    then
+        echo_date "Zone group ${zone_group_name} already opted-in"
+        echo "$zone_name" >> "${SHARED_DIR}/edge-zone-names.txt"
+        echo -en "$zone_group_name" > "${SHARED_DIR}/edge-zone-group-name.txt"
+        return
+    fi
+
+    aws --region "$REGION" ec2 modify-availability-zone-group --group-name "${zone_group_name}" --opt-in-status opted-in
+    echo_date "Zone group ${zone_group_name} opt-in status modified"
+
+    count=0
+    while true; do
+        aws --region "$REGION" ec2 describe-availability-zones --all-availability-zones \
+            --filters Name=zone-type,Values="${zone_type}" Name=zone-name,Values="$zone_name" \
+            | jq -r '.AvailabilityZones[]' |tee /tmp/az.stat
+        
+        if [[ "$(jq -r .OptInStatus /tmp/az.stat)" == "opted-in" ]]; then break; fi
+
+        if [ $count -ge 10 ]; then
+            echo_date "Timeout waiting for zone ${zone_name} attribute OptInStatus==opted-in"
+            exit 1
+        fi
+
+        count=$((count+1))
+        echo_date "Waiting OptInStatus with value opted-in [$count/10]"
+        sleep 30
+    done
+
+    echo_date "Zone group ${zone_group_name} opted-in."
+    echo_date "Saving zone name ${zone_name}"
+    echo "$zone_name" >> "${SHARED_DIR}/edge-zone-names.txt"
+    echo -en "$zone_group_name" > "${SHARED_DIR}/edge-zone-group-name.txt"
+}
+
+
+function select_option() {
+    local zone_type=${1}
+    case ${zone_type} in
+        "local-zone"|"wavelength-zone")
+            echo "Selecting random zone with type: ${zone_type}";
+            select_and_opt_in_zone "${zone_type}";
+            ;;
+        "") echo "Ignoring single zone EDGE_ZONE_TYPE=${zone_type}" ;;
+        *)
+            echo "ERROR: invalid value for variable EDGE_ZONE_TYPE. Got: ${zone_type}, Allowed values: [local-zone | wavelength-zone]";
+            exit 1 ;;
+    esac
+}
+
+function option_multi_zone() {
+    for ZTYPE in $(echo ${EDGE_ZONE_TYPES-}  | tr ',' '\n');
+    do
+        select_option "$ZTYPE";
+    done
+}
+
+function option_single_zone() {
+    select_option "${EDGE_ZONE_TYPE-}";
+}
+
+option_multi_zone
+# Keeping compatibility with EDGE_ZONE_TYPE (deprecated)
+option_single_zone
```

Comment:
@mtulio this introduced a bug here:
1. If `EDGE_ZONE_TYPE="local-zone"`, two local zones will be selected, e.g. `zones: [us-east-1-chi-1a, us-east-1-nyc-1a]
 `, see [this job](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/47045/rehearse-47045-periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-wavelength-fips-f14/1744559598469124096/artifacts/aws-ipi-wavelength-fips-f14/ipi-aws-pre-edge-zones-opt-in/build-log.txt)
2. If `EDGE_ZONE_TYPE="wavelength-zone"`, one local-zone and one wavelength-zone are selected, e.g. `zones: [us-east-1-lim-1a, us-east-1-wl1-chi-wlz-1]
 `,see [this job](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/47045/rehearse-47045-periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-wavelength-fips-f14/1744559598469124096/artifacts/aws-ipi-wavelength-fips-f14/ipi-aws-pre-edge-zones-opt-in/build-log.txt)

So how about:
```
if [[ ${EDGE_ZONE_TYPE} != "" ]]; then
  # Keeping compatibility with EDGE_ZONE_TYPE (deprecated)
  option_single_zone
else
  option_multi_zone
fi
```

Reply:
@yunjiang29 it would be better. I also considering to completely remove the `EDGE_ZONE_TYPE` renaming any reference to `EDGE_ZONE_TYPES`, so we don't need further work in this piece.

Reply:
added the if statement as per your suggestions. I evaluated the completely deprecation (renaming) the env var EDGE_ZONE_TYPE and it will generate double of changes of this PR which is well tested and targeting to merge soon. I will suggest to address it in separated PR after this is merged, [as mentioned by you](https://github.com/openshift/release/pull/47045#pullrequestreview-1810344446). Thanks

Reply:
Sure, I will handle that.